src/main/kotlin/io/github/enyason/predictions/PredictionsApiService.kt

```diff
@@ -29,4 +30,7 @@ interface PredictionsApiService {
 
     @POST("$ENDPOINT/{predictionId}/cancel")
     suspend fun cancelPrediction(@Path("predictionId") predictionId: String): Response<Unit>
+
+    @GET(ENDPOINT)
+    suspend fun listPredictions(@Query("cursor") cursor: String): ResponseBody
```

Comment:
can we design it in such a way that if there is no cursor (that is , it is null) we don't append the cursor query param.
currently from your logic the initial request would look like this:

`https://api.replicate.com/v1/predictions?cursor=`

updating cursor to null, like i suggested earlier would make your First request look like:

`https://api.replicate.com/v1/predictions`